### ISSUE TITLE
No rule for "No overload matches this call"?

### ISSUE THREAD

**Issue by aurelijusrozenas** [typescript-eslint/typescript-eslint#1280]
Hi,

I am not sure if this is the place for this issue and maybe I just cannot google it right.
When compiling I get error `TS2769: No overload matches this call.` but eslint does not produce any errors/warnings.
Am I crazy to expect eslint to catch this, or is there a rule and recommended ruleset does not cover it?

Thanks!

```TS
// code to compile
const myListener = function (variableWithWrongType: string): void {
    console.log(variableWithWrongType);
};
window.addEventListener('load', myListener);
```

```JS
// .eslintrc.js
module.exports = {
    parser: '@typescript-eslint/parser',
    extends: [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended"
    ],
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.9.0` |
| `@typescript-eslint/parser`        | `2.9.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
We don't emit typescript compiler errors, nor do we generally build rules which duplicate compiler errors.
We generally look to _augment_ the checks the compiler does.

In this instance - a lint rule wouldn't make sense, as it's a direct feature of the compiler.
